web/src/lib/utils/file-uploader.ts

```diff
@@ -46,25 +45,20 @@ export const fileUploadHandler = async (
 	sharedKey: string | undefined = undefined,
 	onDone?: (id: string) => void
 ) => {
-	if (files.length > 50) {
-		notificationController.show({
-			type: NotificationType.Error,
-			message: `Cannot upload more than 50 files at a time - you are uploading ${files.length} files. 
-			Please check out <u>the bulk upload documentation</u> if you need to upload more than 50 files.`,
-			timeout: 10000,
-			action: { type: 'link', target: 'https://immich.app/docs/features/bulk-upload' }
-		});
-
-		return;
-	}
-
+	const files$ = new Subject<File>();
+	files$
+		.pipe(
+			mergeMap(async (file) => {
+				await fileUploader(file, albumId, sharedKey, onDone);
+			}, 2)
+		)
+		.subscribe();
 	const acceptedFile = files.filter((file) => {
 		const assetType = getFileMimeType(file).split('/')[0];
 		return assetType === 'video' || assetType === 'image';
 	});
-
-	for (const asset of acceptedFile) {
-		await fileUploader(asset, albumId, sharedKey, onDone);
+	for (const file of acceptedFile) {
```

Comment:
The lines above this also loop through the list of assets. Can we combine the two loops into one? ie for each asset, get mime type, if true add to `files$`?